org.eclipse.jdt.core/formatter/org/eclipse/jdt/internal/formatter/CommentsPreparator.java

```diff
@@ -793,6 +794,238 @@ private void alignJavadocTag(List<Token> tagTokens, int paramNameAlign, int desc
 		}
 	}
 
+	private void handleMarkdown(TagElement node) {
+		if (!(node.getParent() instanceof Javadoc javaDoc) || !javaDoc.isMarkdown()
+				|| !this.options.comment_format_markdown_comment) {
+			return;
+		}
+
+		String text = this.tm.toString(node);
+
+		Matcher matcher = MARKDOWN_HEADINGS_PATTERN_1.matcher(text); // Check for MarkDown headings #h1 - #h6
+		while (matcher.find()) {
+			int startPos = matcher.start() + node.getStartPosition();
+			int tokenIndex = tokenStartingAt(startPos);
+			Token headingToken = this.ctm.get(tokenIndex);
+			if (tokenIndex != 1) {
+				headingToken.putLineBreaksBefore(2);
+			}
+			int endPos = matcher.end() + node.getStartPosition();
+			int endIndex = tokenEndingAt(endPos - 1);
+			Token endingToken = this.ctm.get(endIndex);
+			endingToken.breakAfter();
+			headingToken.spaceBefore();
+		}
+
+		matcher = MARKDOWN_HEADINGS_PATTERN_2.matcher(text); // Check for MarkDown headings with styles '-- & ==='
+		while (matcher.find()) {
+			int startPos = matcher.start() + node.getStartPosition();
+			int tokenIndex = tokenStartingAt(startPos);
+			Token headingToken = this.ctm.get(tokenIndex);
+			if (tokenIndex != 1) {
+				headingToken.breakBefore();
+			}
+			headingToken.breakAfter();
+		}
+
+		List<ASTNode> fragments = node.fragments();
+		ArrayDeque<ASTNode> fragmentsDeque = new ArrayDeque<>(fragments);
+		handleMarkdownList(fragmentsDeque, 1, 0, null);
+		handleMarkdownTable(fragments);
+
+		matcher = MARKDOWN_FENCES_PATTERN.matcher(text); // Check for Markdown snippet with styles '``` & ```'
+		while (matcher.find()) {
+			int startPos = matcher.start() + node.getStartPosition();
+			int tokenIndex = this.ctm.findIndex(startPos, ANY, true);
+			Token openingToken = this.ctm.get(tokenIndex);
+			String openingFence = this.ctm.toString(openingToken);
+			int endPos = -1;
+			int tokenIndexLast = -1;
+			Token closingToken = null;
+			String closingSnippet = null;
+			boolean shouldDisable = true;
+
+			while (matcher.find()) {
+				endPos = matcher.start() + node.getStartPosition();
+				tokenIndexLast = this.ctm.findIndex(endPos, ANY, true);
+				closingToken = this.ctm.get(tokenIndexLast);
+				closingSnippet = this.ctm.toString(closingToken);
+
+				// Check if fences match
+				if (!openingFence.replaceAll("[^`~]", "").equals(closingSnippet)) //$NON-NLS-1$ //$NON-NLS-2$
+					continue;
+
+				long openCount = openingFence.chars().filter(ch -> ch == openingFence.charAt(0)).count();
+				long closeCount = closingSnippet.chars().filter(ch -> ch == openingFence.charAt(0)).count();
+
+				if (openCount == closeCount) {
+					if (tokenIndex > 1)
+						openingToken.breakBefore();
+					openingToken.breakAfter();
+					if (this.ctm.size() - 1 != tokenIndexLast) {
+						closingToken.putLineBreaksAfter(1);
+					}
+					String languageMarker = openingFence.replaceAll("[`~]", ""); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
this is called the info string in commonmark spec. It doesn't have to be in the same token, can be separated by spaces/tabs from the ticks/tildas